system/core/HyphaRequest.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+class HyphaRequest {
+	/** TODO[LRM]: allow registration of system pages, then inject system pages in constructor */
+	const HYPHA_SYSTEM_PAGE_FILES = 'files';
+	const HYPHA_SYSTEM_PAGE_IMAGES = 'images';
+	const HYPHA_SYSTEM_PAGE_INDEX = 'index';
+	const HYPHA_SYSTEM_PAGE_SETTINGS = 'settings';
+	const HYPHA_SYSTEM_PAGE_UPLOAD = 'upload';
+	const HYPHA_SYSTEM_PAGE_CHOOSER = 'chooser';
+
+	/** @var string */
+	private $requestQuery;
+
+	/** @var array */
+	private $isoLangList;
+
+	/** @var array */
+	private $requestParts;
+
+	/** @var null|false|string */
+	private $language;
+
+	/** @var null|false|string */
+	private $systemPage;
+
+	/**
+	 * @param $requestQuery
+	 * @param array $isoLangList Associative array containing connecting iso639 language codes with their native name (and english translation)
+	 */
+	public function __construct($requestQuery, array $isoLangList) {
+		$this->requestQuery = $requestQuery;
+		$this->isoLangList = $isoLangList;
+	}
+
+	public static function getSystemPages() {
+		return [
+			HyphaRequest::HYPHA_SYSTEM_PAGE_FILES,
+			HyphaRequest::HYPHA_SYSTEM_PAGE_IMAGES,
+			HyphaRequest::HYPHA_SYSTEM_PAGE_INDEX,
+			HyphaRequest::HYPHA_SYSTEM_PAGE_SETTINGS,
+			HyphaRequest::HYPHA_SYSTEM_PAGE_UPLOAD,
+			HyphaRequest::HYPHA_SYSTEM_PAGE_CHOOSER,
+		];
+	}
+
+	/**
+	 * @param bool $excludeLanguage
+	 *
+	 * @return string
+	 */
+	public function getRequestQuery($excludeLanguage = true) {
+		return implode('/', $this->getRequestParts($excludeLanguage));
+	}
+
+	/**
+	 * @param bool $excludeLanguage
+	 *
+	 * @return array
+	 */
+	public function getRequestParts($excludeLanguage = true) {
```

Comment:
Would it make sense to also remove the system pagename and page type from this list? Or perhaps add a new `getArgs()` method that returns all the "extra" request parts (e.g. everything that is not accessible through the other get* methods)?

This would allow system page loading below to handle `$args[0]`, now it skips the first element since that is the system page type (which is a bit of an abstraction leak).

Reply:
Agreed, I'll create a `getArgs()` method for this.